### PR TITLE
core: several follow-ups for ExtraFileDescriptors

### PR DESCRIPTION
**Pull request by YHNdnzj**

### CODE REVIEW

File: src/core/dbus-service.c [hunk 1]
```diff
@@ -767,29 +765,30 @@ static int bus_service_set_transient_property(
                         /* Disallow empty string for ExtraFileDescriptors.
                          * Unlike OpenFile, StandardInput and friends, there isn't a good sane
                          * default for an arbitrary FD. */
-                        if (fd < 0 || isempty(fdname) || !fdname_is_valid(fdname))
```
Comment (ryantimwilson) [inline src/core/dbus-service.c:770]: Any reason we removed the fd < 0 check?

File: src/core/service.c [hunk 2]
```diff
@@ -3181,6 +3144,21 @@ static int service_serialize(Unit *u, FILE *f, FDSet *fds) {
                 (void) serialize_item_format(f, "fd-store-fd", "%i \"%s\" %s", copy, c, one_zero(fs->do_poll));
         }
 
+        FOREACH_ARRAY(i, s->extra_fds, s->n_extra_fds) {
```
Comment (ryantimwilson) [inline src/core/service.c:3147]: Thanks for catching this! Sorry I forgot this. Seems this is used when PID1/service manager restarts in order to not lose state (let me know if that's wrong). If so, I can add an integration test for this once your PR merges

**Review by ryantimwilson — APPROVED** [review]

File: src/core/dbus-service.c (hunk 1 above)
Comment (YHNdnzj) [reply src/core/dbus-service.c:770]: dbus's native `h` type would never return invalid fds to us. So this isn't necessary.

File: src/core/service.c (hunk 2 above)
Comment (YHNdnzj) [reply src/core/service.c:3147]: Yes, please :)

**Review by yuwata — APPROVED** [review]
LGTM.

File: src/core/service.c (hunk 2 above)
Comment (ryantimwilson) [reply src/core/service.c:3147]: @YHNdnzj https://github.com/systemd/systemd/pull/34733

Thanks again!

**Pull request merged**